pkg/ceph/client/pool.go

```diff
@@ -129,20 +130,33 @@ func CreatePool(context *clusterd.Context, clusterName string, newPool CephStora
 }
 
 func CreatePoolForApp(context *clusterd.Context, clusterName string, newPool CephStoragePoolDetails, appName string) error {
-	args := []string{"osd", "pool", "create", newPool.Name, strconv.Itoa(newPool.Number)}
+	// create a crush rule for a replicated pool, if a failure domain is specified
+	replicated := newPool.ErasureCodeProfile == "" && newPool.Size > 0
+	ruleName := fmt.Sprintf("%s-rule", newPool.Name)
+	if replicated && newPool.FailureDomain != "" {
+		args := []string{"osd", "crush", "rule", "create-simple", ruleName, "default", newPool.FailureDomain}
+		_, err := ExecuteCephCommand(context, clusterName, args)
+		if err != nil {
+			return fmt.Errorf("failed to create pool %s. %+v", newPool.Name, err)
```

Comment:
copy/paste error message?  this failure is for creating a crush rule, not a pool

Reply:
good catch